File: 1051-height-checker/1051-height-checker.js
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function(heights) {
    var dontMatch = 0;
    var expected = heights.slice();
    expected.sort(function(a, b) {
  return a - b;
});
    for(var i =0; i<heights.length; i++){
        if(heights[i] != expected[i]) {
            dontMatch++;
        }
    }
    return dontMatch;
};

File: 1051-height-checker/1051-height-checker.java


class Solution {
    public int heightChecker(int[] heights) {
        int dontMatch = 0;
        int[] expected = Arrays.copyOf(heights, heights.length); // Create a copy of the heights array
        Arrays.sort(expected); // Sort the expected array
        for (int i = 0; i < heights.length; i++) { // Use heights.length as property
            if (heights[i] != expected[i]) dontMatch++;
        }
        return dontMatch;
    }
}
